src/storage.cpp

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include "nvram.hpp"
+#include "storage.hpp"
+
+#include <phosphor-logging/log.hpp>
+
+#include <exception>
+
+using namespace phosphor::logging;
+
+// Special variable which contains default values for UEFI settings
+static const VariableKey stdDefaults{"StdDefaults",
+                                     {0x45, 0x99, 0xD2, 0x6F, 0x1A, 0x11, 0x49,
+                                      0xB8, 0xB9, 0x1F, 0x85, 0x87, 0x45, 0xCF,
+                                      0xF8, 0x24}};
+
+Storage::Storage(const std::filesystem::path& varFile) : file(varFile)
+{
+    if (!std::filesystem::exists(file))
+    {
+        log<level::WARNING>("UEFI storage is empty",
+                            entry("FILE=%s", file.c_str()));
+    }
+    else
+    {
+        variables = loadVariables(file);
+        log<level::INFO>("UEFI settings loaded", entry("FILE=%s", file.c_str()),
+                         entry("VARS=%u", variables.size()));
+    }
+}
+
+bool Storage::empty() const
+{
+    return variables.empty();
+}
+
+std::optional<VariableValue> Storage::get(const VariableKey& key)
+{
+    std::unique_lock<std::mutex> lock(mutex);
```

Comment:
I guess it's not necessary as the app is single threading.

Reply:
Fixed